sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/CertificateValidationCheckBase.cs

```diff
@@ -0,0 +1,298 @@
+                    && parameters[3].Type.Is(KnownType.System_Net_Security_SslPolicyErrors)
+                    && namedSymbol.DelegateInvokeMethod.ReturnType.Is(KnownType.System_Boolean);
+            }
+            return false;
+        }
+
+        private ImmutableArray<Location> ArgumentLocations(InspectionContext c, TExpressionSyntax expression)
+        {
+            switch (TryExtractAddressOfOperand(expression))
+            {
+                case TIdentifierNameSyntax identifier:
+                    var identSymbol = c.Context.SemanticModel.GetSymbolInfo(identifier).Symbol;
+                    if (identSymbol != null && identSymbol.DeclaringSyntaxReferences.Length == 1)
+                    {
+                        return IdentifierLocations(c, SyntaxFromReference(identSymbol.DeclaringSyntaxReferences.Single()));
+                    }
+                    break;
+                case TLambdaSyntax lambda:
+                    return LambdaLocations(c, lambda);
+                case TInvocationExpressionSyntax invocation:
+                    var invSymbol = c.Context.SemanticModel.GetSymbolInfo(invocation).Symbol;
+                    if (invSymbol != null && invSymbol.DeclaringSyntaxReferences.Length == 1 && SyntaxFromReference(invSymbol.DeclaringSyntaxReferences.Single()) is TMethodSyntax syntax)
+                    {
+                        c.VisitedMethods.Add(syntax);
+                        return InvocationLocations(c, syntax);
+                    }
+                    break;
+            }
+            return ImmutableArray<Location>.Empty;
+        }
+
+        private ImmutableArray<Location> IdentifierLocations(InspectionContext c, SyntaxNode syntax)
+        {
+            switch (syntax)
+            {
+                case TMethodSyntax method:                  //Direct delegate name
+                    return BlockLocations(c, method);  
+                case TParameterSyntax parameter:            //Value arrived as a parameter
+                    return ParamLocations(c, parameter);
+                case TVariableSyntax variable:     //Value passed as variable
+                    return VariableLocations(c, variable);
+            }
+            return ImmutableArray<Location>.Empty;
+        }
+
+        protected ImmutableArray<Location> ParamLocations(InspectionContext c, TParameterSyntax param)
+        {
+            var ret = ImmutableArray.CreateBuilder<Location>();
+            var containingMethodDeclaration = param.FirstAncestorOrSelf<TMethodSyntax>();
+            if (!c.VisitedMethods.Contains(containingMethodDeclaration))
+            {
+                c.VisitedMethods.Add(containingMethodDeclaration);
+                var containingMethod = c.Context.SemanticModel.GetDeclaredSymbol(containingMethodDeclaration) as IMethodSymbol;
+                var paramSymbol = containingMethod.Parameters.Single(x => x.Name == IdentifierText(param));
```

Comment:
You can extract `IdentifierText(param)` outside the iteration as it will always have the same value